### PR TITLE
[DLSP21] 11-2 speech recognition

### PR DESCRIPTION
**Pull request by gyaneshg96**
3rd quarter of the lecture

### CODE REVIEW

**Review by Atcold — CHANGES_REQUESTED** [review]
This looks good. Please, use `\vect{}` for vectors throughout the write up and add the missing part.

File: .gitmodules [hunk 1]
```diff
@@ -1,3 +1,3 @@
 [submodule "docs/jekyllbook"]
 	path = docs/jekyllbook
-	url = https://github.com/ebetica/jekyllbook
+	url = git@github.com:ebetica/jekyllbook.git
```
Comment (Atcold) [inline .gitmodules:3]: Undo this.

**Review by Atcold — CHANGES_REQUESTED** [review]

File: .gitmodules (hunk 1 above)
Comment (gyaneshg96) [reply .gitmodules:3]: Okay

Comment (Atcold) [reply .gitmodules:3]: It's still here.

File: docs/en/week11/11-2.md [hunk 2]
```diff
@@ -0,0 +1,382 @@
...
+* Facilitate research: When we separate the data from the code it makes it easier to explore different ideas. In our case graph is the data and the code is operations allows us to explore different ideas by changing the graph.
+
+
+## Sequence Criteria with WFSAs
+
+It turns out that lots of the sequence criteria like connection temporal classification (CTC) can be specified as the difference between two forward scores.
+
+* Graphs A: function of input X (speech) and target "output" Y (transription).
+* Graph Z: function of input X, which servers as normalization.
+* Loss: $$logP(Y \mid X) = forwardScore(A_{X,Y}) - forwardScore(Z_{X})$$
```
Comment (Atcold) [inline docs/en/week11/11-2.md:92]: `log` wants a `\` before.
Use `\text{}` for undefined function names.

File: docs/en/week11/11-2.md [hunk 3]
```diff
@@ -0,0 +1,382 @@
...
+* Encode Priors: easy to encode priors into a graph
+* End-to-end: Bridge the training and test process 
+* Facilitate research: When we separate the data from the code it makes it easier to explore different ideas. In our case graph is the data and the code is operations allows us to explore different ideas by changing the graph.
+
+
+## Sequence Criteria with WFSAs
+
+It turns out that lots of the sequence criteria like connection temporal classification (CTC) can be specified as the difference between two forward scores.
+
+* Graphs A: function of input X (speech) and target "output" Y (transription).
```
Comment (Atcold) [inline docs/en/week11/11-2.md:90]: Use `$` for math, here and everywhere else.

File: docs/en/week11/11-2.md [hunk 4]
```diff
@@ -0,0 +1,382 @@
...
+#### Example
+
+A simple WFSA which recognizes aa or ab
+* The score of **aa** is 0 + 2 = 2
+* The score of **ba** is 1 + 2 = 3
+
+
+<center>
+<img src="{{site.baseurl}}/images/week11/11-2/figure1.png" style="background-color:#DCDCDC;"/><br>
+<b>Figure 1:</b> WFSA<br>
```
Comment (Atcold) [inline docs/en/week11/11-2.md:131]: This doesn't seem to be figure 1…
Also, there's no caption. Only a cryptic WFSA. Add an actual caption, description of the image. Here and elsewhere.

File: docs/en/week11/11-2.md [hunk 5]
```diff
@@ -0,0 +1,382 @@
...
+<img src="{{site.baseurl}}/images/week11/11-2/figure10.png" style="background-color:#DCDCDC;"/><br>
+<b>Figure 10:</b> Target Constrained Graph <br>
+<br>
+</center>
+
+
+### Loss
+
+$$
+loss = - (forwardScore(A) - forwardScore(E))
```
Comment (Atcold) [inline docs/en/week11/11-2.md:262]: "loss" is not a mathematical symbol. Variables have a single character. Use `\ell` with correct arguments.
Also, use `\text{}` for the function names.

File: docs/en/week11/11-2.md [hunk 6]
```diff
@@ -0,0 +1,382 @@
...
+
+
+## Code examples:
+
+### Make the target graph
+
+Similar to figure 8.
+
+~~~
+
```
Comment (Atcold) [inline docs/en/week11/11-2.md:275]:
```suggestion
~~~python
```

Here and below.

File: docs/en/week11/11-2.md [hunk 7]
```diff
@@ -0,0 +1,382 @@
...
+# Add arcs:
+target.add_arc(src_node=0, dst_node=1, label=0)
+target.add_arc(src_node=1, dst_node=1, label=0)
+target.add_arc(src_node=1, dst_node=2, label=1)
+target.add_arc(src_node=2, dst_node=2, label=1)
+# Draw the graph
+label_map = {0: 'a', 1: 'b'}
+gtn.draw(target, "target.pdf", label_map)
+
+~~~
```
Comment (Atcold) [inline docs/en/week11/11-2.md:293]:
```suggestion
~~~
```

Here and below.

**Review by Atcold — CHANGES_REQUESTED** [review]
Good. A few edits more.

Add 50–60 word summary to `12.md`. Right now it's too short.
Invert all the figures, so they are white text on black background (shouldn't be hard).
Improve caption and fix caption numbers.

File: docs/en/week11/11-2.md (hunk 3 above)
Comment (abdulrahimq) [reply docs/en/week11/11-2.md:90]: Just to confirm the whole document 11-2.md should have $ instead of $$ correct? Because in the contribution guide it says that we should use $$.

File: docs/en/week11/11-2.md [hunk 8]
```diff
@@ -0,0 +1,382 @@
...
+#### Example
+
+A simple WFSA which recognizes aa or ab
+* The score of **aa** is 0 + 2 = 2
+* The score of **ba** is 1 + 2 = 3
+
+
+<center>
+<img src="{{site.baseurl}}/images/week11/11-2/figure1.png" style="background-color:#DCDCDC; filter: invert(100%)"/><br>
+<b>Figure 1:</b> Three Nodes Weighted Finite-State Acceptor (WFSA)<br>
```
Comment (Atcold) [inline docs/en/week11/11-2.md:131]: I don't think this is figure **1**.
Same for the captions below.

Please, submit a consistent write up.

File: docs/en/week11/11-2.md (hunk 6 above)
Comment (Atcold) [reply docs/en/week11/11-2.md:275]: Can we please use syntax highlighting? Here and below.
Also, remove the empty lines before the first line of code and after the last line of code.

Thanks.

**Review by Atcold — CHANGES_REQUESTED** [review]
Almost there. Please, read **all** the feedback carefully.
Let's try to finish this up.

Also, you need to write a 50–60 word summary inside `11.md`. Now it's a little too short.

File: docs/en/week11/11-2.md (hunk 8 above)
Comment (abdulrahimq) [reply docs/en/week11/11-2.md]: I'm not sure I fully sure I understand what you mean here. I'm assuming you mean it is figure 5 and I need to change it to Figure 5: in the caption. if not please let me know.

**Comment by Atcold** [comment]
Thanks!
